config/routes.rb

```diff
@@ -1,6 +1,13 @@
-Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+# frozen_string_literal: true
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+Rails.application.routes.draw do
+  root 'users#index'
+  get '/users', to: 'users#index', as: :users
+  get '/users:id', to: 'users#show', as: :user
+  get 'users:user_id/posts', to: 'posts#index', as: :user_posts
+  get 'users:user_id/posts:id', to: 'posts#show', as: :user_post
+  get '/posts/new', to: 'posts#new', as: :new_user_post
+  post '/posts/new', to: 'posts#create' #create_new_post
+  post 'comments/new', to: 'comments#create', as: :new_user_comment #create_new_comment
+  post 'likes/new', to: 'likes#create', as: :new_user_like #create_new_like
 end
```

Comment:
- [ ] You have made all routes but it would be nice if you make them using the new syntax of Rails as shown below :arrow_down:

```ruby
Rails.application.routes.draw do
  root "users#index"
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create, :destroy] do
      resources :comments, only: [:new, :create, :destroy]
      resources :likes, only: [:create]
    end
  end
end
`
```

Reply:
@HAMMAS-SALEEM this should be marked as an optional comment not to force me to do it!!!!

Reply:
@shakerAbuDrais This requirement cannot be requested optionally because the previous reviewer asked you to make routes for new and create methods of controllers so if you are making something new so i would suggest you to make it in the right way. I hope you can understand :+1: If you don't understand or want to discuss then you can reach out to me on slack by searching mu name Hammas in full-time-students channel :+1: